### DISCUSSION TITLE
[Question] - Upgrade path to 1.3.x fron 1.2.1

### DISCUSSION THREAD

**Discussion by RegisWeduLabs** [harvester/harvester#6009] [General]
Hi,

I posted this discussion a few weeks back https://github.com/harvester/harvester/discussions/5826 - but I would like to get clarity specifically on this:

Roadmap Clarity: The roadmap (https://github.com/harvester/harvester/wiki/Roadmap) shows that version 1.2.2 was released in May, suggesting an upgrade path of 1.2.1->1.2.2->1.3.1. However, there is also a 1.2.3 release listed in the milestones for July after 1.3.1 releases in June. If the upgrade path is 1.2.1->1.2.2->1.3.1, why is there a 1.2.3 release, and what does it mean for the upgrade path? Does this indicate a "fork" in the upgrade path, similar to Rancher, where there are parallel releases? This information is crucial for planning upgrades, especially since we need to schedule maintenance and inform users weeks in advance. It is also important to understand the potential security risks for Harvester users in delaying an upgrade to combine 2-3 releases in one go.

Basically, I need to plan the upgrade to 1.3.x - but I am confused on how to go about it.  I do not want to schedule 2 different days for the upgrade.  Is it possible to give us a better understanding of the upgrade path to 1.3.x from 1.2.1?  From what I understand, there are 2 scenarios:

1) 1.2.1=>1.2.2=>1.3.2
2) 1.2.1=>1.2.2=>1.2.3=>1.3.2 (or 1.2.1=>1.2.3=>1.3.2)

Is it possible to clarify what the upgrade path will be please?

Thank you

**Comment by bk201** [comment]
The v1.2.3 and v1.3.2's date are still to be defined.

We'll keep releasing patch releases for supported versions (now it's v1.2.x and v1.3.x).
In general, 1.2.x is considered stable. 1.3.x is newer and has newer features. (you can check this [page](https://www.suse.com/lifecycle/#harvester) for Harvester lifecycle). If you don't need new features in Harvester v1.3.x and stability is the concern, you can consider staying in v1.2.x releases. But when v1.2.x is near its EoL, you can plan to upgrade Harvester to v1.3.x (there will be a tested upgrade path from v1.2 to v1.3 at the time).

**Comment by RegisWeduLabs** [comment]
Hey @bk201, thank you for your reply.

1.3.1 just released, and I see that we can now upgrade from 1.2.2 -> 1.3.1, so the upgrade path is 1.2.1->1.2.2->1.3.1 to get to the latest version.  That's now clear in my mind and I can plan accordingly.  

Again, thank you for clarifying this.

Regards
